packages/truffle-db/src/workspace/test/index.ts

```diff
@@ -313,11 +358,11 @@ it("adds compilation", async () => {
     compilerName: Migrations.compiler.name,
     compilerVersion: Migrations.compiler.version,
     sourceId: soliditySha3(Migrations.source, Migrations.sourcePath),
-    id: "0xfd63faebf4e7515b2f48fdf22066869e09df9b2e91f4e0cf88d7d14ec496b515", 
+    id: soliditySha3(jsonStableStringify(Migrations.compiler, soliditySha3(Migrations.source, Migrations.sourcePath))), 
     contractId: soliditySha3(Migrations.contractName, soliditySha3(Migrations.source, Migrations.sourcePath))
   }
 
-  const compilationId = soliditySha3(variables.contractId, variables.sourceId)
+  const compilationId = soliditySha3(jsonStableStringify(Migrations.compiler, soliditySha3(Migrations.source, Migrations.sourcePath)))
```

Comment:
ID calculations getting outta hand!

Some thoughts... firstly, it feels cleaner to add wrapping braces to the IDs, a là
```suggestion
  const compilationId = soliditySha3(jsonStableStringify(Migrations.compiler, soliditySha3(Migrations.source, Migrations.sourcePath)}))
```

But moreover, it gets rather brittle to have these ID re-computations. I think we should only compute ID when that's the thing we're trying to test. So if we're testing `sourcesAdd`, then we should test that the source ID matches our expectation.

Beyond that (e.g. to avoid needing to compute compilation IDs inside the contract tests), I think we should work on getting helper methods in place so it becomes easier to run pre-cursor mutations and get the IDs that way. If you're into the work I did in #1918, we can merge that and then start looking at using `describe()` blocks with `before()` hooks to add these requisite dependent resources (e.g.. `describe("Compilation", ...)` could have `before("adds sources", () => { })`, etc.). What do you think?